### ISSUE TITLE
Add more tests for RadioButtonGroup and CheckBoxGroup

### ISSUE THREAD

**Issue by elmot** [vaadin/framework8-issues#291]
add tests for disabled group not being selectable and being selectable again when enabled

add tests for verifying that html content is displayed when used

missing test for custom itemCaptionProdiver and itemIconProvider that verifies that those even work

**Comment by denis-anisimov** [comment]
Blocked by #286.
The tests can be implemented but they will fail until #286 is fixed.

**Comment by Legioth** [comment]
Tests for problems described in #286 can be added done separately

**Comment by denis-anisimov** [comment]
https://dev.vaadin.com/review/14452

Tests for DOM when disabled will be in the separate issue (part of #286).

Item Icon Provider doesn't work. It's #301. It will contain tests for that.

In the end this issue is about : selection active/inactive for enabled/disabled component and ItemCaptionProvider.

**Comment by denis-anisimov** [comment]
https://dev.vaadin.com/review/#/c/14452/
Patch contains also ItemIconProvider test for CheckBoxGroup where it works correctly.
